### PR TITLE
Split API part out from core, into a separate core-api package

### PR DESCRIPTION
**Pull request by Rugvip**
Wanting this split because for a couple of reasons, the main one being that core currently depends on test-utils, which is blocking us from creating good wrappers for rendering components inside test apps. It also makes sure that the apis don't end up with dependencies on components, which is something we want to avoid.

`@backstage/core` reexports most things from `@backstage/core-api`, so there's no need to add it as a separate dependency, i.e. this change should not be noticed from the outside except for the extra install.

### CODE REVIEW

File: packages/core-api/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# @backstage/core-api
+
+This package provides the core API used by Backstage plugins and apps.
+
+## Installation
+
+Do not install this package directly, it is reexported by `@backstage/core`. Install that instead:
+
+```sh
+$ npm install --save @backstage/core
```
Comment (freben) [inline packages/core-api/README.md:10]: should we even mention the npm possibility?

File: packages/core-api/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# @backstage/core-api
+
+This package provides the core API used by Backstage plugins and apps.
+
+## Installation
+
+Do not install this package directly, it is reexported by `@backstage/core`. Install that instead:
```
Comment (freben) [inline packages/core-api/README.md:7]: link out to the core folder on github?

File: packages/core-api/src/app/App.tsx [hunk 3]
```diff
@@ -17,19 +17,13 @@
 import React, { ComponentType, FC } from 'react';
```
Comment (freben) [inline packages/core-api/src/app/App.tsx:17]: could perhaps even name this file `PrivateAppImpl.tsx` if you wanted to

File: packages/core/src/index.ts [hunk 4]
```diff
@@ -14,6 +14,8 @@
  * limitations under the License.
  */
 
+export * from '@backstage/core-api';
```
Comment (freben) [inline packages/core/src/index.ts:17]: `export * from '*';`

**Review by freben — APPROVED** [review]

File: packages/core-api/README.md (hunk 1 above)
Comment (Rugvip) [reply packages/core-api/README.md:10]: Meh, people can use npm if they want. You could set up a new repo, or just switch to npm after scaffolding an app

File: packages/core/src/index.ts (hunk 4 above)
Comment (Rugvip) [reply packages/core/src/index.ts:17]: wat?

File: packages/core-api/src/app/App.tsx (hunk 3 above)
Comment (Rugvip) [reply packages/core-api/src/app/App.tsx:17]: Yeah maybe, not this PR though imo

File: packages/core/src/index.ts (hunk 4 above)
Comment (freben) [reply packages/core/src/index.ts:17]: just kidding :)

**Pull request merged**